app/controllers/concerns/find_resource.rb

```diff
@@ -9,8 +9,19 @@ def resource_id
     [ params[:account], params[:kind], params[:identifier] ].join(":")
   end
 
-  def find_resource
-    @resource ||= Resource.find_if_visible current_user, resource_id: resource_id
-    raise Exceptions::RecordNotFound, resource_id unless @resource
+  def resource!
```

Comment:
What's the reason for the bang on this method name?

Reply:
Probably because it doesn't do any visibility checking.

Reply:
Yup, I deliberated for a good couple minutes on the right name for the method, until I remembered the Ruby convention of having bang methods being the counterpart of non-bang, only more dangerous. Seemed fitting.